app/services/trainees/create_from_dttp.rb

```diff
@@ -0,0 +1,333 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromDttp
+    include ServicePattern
+
+    class UnrecognisedStatusError < StandardError; end
+
+    def initialize(dttp_trainee:)
+      @dttp_trainee = dttp_trainee
+      @trainee = Trainee.new(mapped_attributes)
+    end
+
+    def call
+      return if dttp_trainee.provider.blank?
+      return if latest_placement_assignment.blank?
+
+      if training_route.blank?
+        dttp_trainee.non_processable_missing_route!
+        return
+      end
+
+      if training_route == TRAINING_ROUTE_ENUMS[:hpitt_postgrad]
+        dttp_trainee.non_processable_hpitt!
+        return
+      end
+
+      if trainee_already_exists?
+        dttp_trainee.non_processable_duplicate!
+        return
+      end
+
+      trainee.save!
+      create_degrees!
+
+      # TODO: TBC what states are we going to use
+      dttp_trainee.processed!
+
+      trainee
+    end
+
+  private
+
+    attr_reader :dttp_trainee, :trainee
+
+    def mapped_attributes
+      return if latest_placement_assignment.blank?
+
+      {
+        state: trainee_status,
+        provider: dttp_trainee.provider,
+        trainee_id: trainee_id,
+        training_route: training_route,
+        trn: dttp_trainee.response["dfe_trn"],
+        submitted_for_trn_at: latest_placement_assignment.response["dfe_trnassessmentdate"],
```

Comment:
When there are multiple placement assignments does this get copied to the later ones? I expect it would be sent on  the first rather than the last.

Reply:
Happy to change to first, but would be worth discussing about which placement assingment to use at our chat today.